client/my-sites/pages/page/index.js

```diff
@@ -421,42 +435,155 @@ class Page extends Component {
 						siteUrl={ this.props.multisite && this.getSiteDomain() }
 					/>
 				</div>
-				{ ellipsisGridicon }
-				{ popoverMenu }
-				<ReactCSSTransitionGroup
-					transitionName="updated-trans"
-					transitionEnterTimeout={ 300 }
-					transitionLeaveTimeout={ 300 }
-				>
-					{ this.props.buildUpdateTemplate() }
-				</ReactCSSTransitionGroup>
+				{ ellipsisMenu }
+				{ this.renderShadowNotice() }
 			</CompactCard>
 		);
 	}
 
+	async onShadow( status ) {
+		await this.props.onShadow( this.props.page.global_ID, !! status );
+		return await new Promise( resolve => this.setState( { shadowStatus: status }, resolve ) );
+	}
+
+	async performUpdate( { action, progressNotice, successNotice, errorNotice, undo } ) {
+		await this.onShadow( progressNotice );
+		try {
+			await action();
+			if ( undo === false ) {
+				await this.onShadow( false );
+				return;
+			}
+			await this.onShadow( { ...successNotice, undo } );
+			if ( ! undo ) {
+				await sleep( 5000 );
+				await this.onShadow( false );
+			}
+		} catch ( error ) {
+			await this.onShadow( errorNotice );
+			await sleep( 5000 );
+			await this.onShadow( false );
+		}
+	}
+
+	async updatePostStatus( status ) {
+		const { page, translate } = this.props;
+
+		switch ( status ) {
+			case 'delete':
+				await this.performUpdate( {
+					action: () => this.props.deletePost( page.site_ID, page.ID ),
+					progressNotice: {
+						status: 'is-error',
+						icon: 'trash',
+						text: translate( 'Deleting…' ),
+					},
+					successNotice: {
+						status: 'is-success',
+						text: translate( 'Page deleted.' ),
+					},
+					errorNotice: {
+						status: 'is-error',
+						text: translate( 'Failed to delete page.' ),
+					},
+				} );
+				return;
+
+			case 'trash':
+				await this.performUpdate( {
+					action: () => this.props.trashPost( page.site_ID, page.ID, page ),
+					undo: page.status !== 'trash' && 'restore',
+					progressNotice: {
+						status: 'is-error',
+						icon: 'trash',
+						text: translate( 'Trashing…' ),
+					},
+					successNotice: {
+						status: 'is-success',
+						text: translate( 'Page trashed' ),
+					},
+					errorNotice: {
+						status: 'is-error',
+						text: translate( 'Failed to trash page.' ),
+					},
+				} );
+				return;
+
+			case 'restore':
+				await this.performUpdate( {
+					action: () => this.props.restorePost( page.site_ID, page.ID ),
+					undo: page.status === 'trash' && 'trash',
+					progressNotice: {
+						status: 'is-warning',
+						icon: 'history',
+						text: translate( 'Restoring…' ),
```

Comment:
Hi! I've found a possible matching string that has already been translated **29** times:
`translate( 'Restoring' )` *ES Score: 23*
See 1 additional suggestion in the [PR translation status page](https://translate.wordpress.com/localci/status/automattic/wp-calypso/remove/update-post-status)

Help me improve these suggestions: react with :thumbsdown: if the suggestion doesn't make any sense, or with :thumbsup: if it's a particularly good one (even if not implemented).